okio/src/jsMain/kotlin/okio/fs.kt

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (C) 2020 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This class declares the subset of Node.js filesystem APIs that we need in Okio.
+ *
+ *
+ * Why not Dukat?
+ * --------------
+ *
+ * This file does manually what ideally [Dukat] would do automatically.
+ *
+ * Dukat's generated stubs need awkward call sites to disambiguate overloads. For example, to call
+ * `mkdirSync()` we must specify an options parameter even though we just want the default:
+ *
+ *   mkdirSync(dir.toString(), options = undefined as MakeDirectoryOptions?)
+ *
+ * By defining our own externals, we can omit the unwanted optional parameter from the declaration.
+ * This leads to nicer calling code!
+ *
+ *   mkdirSync(dir.toString())
+ *
+ * Dukat also gets the nullability wrong for `Dirent.readSync()`.
+ *
+ *
+ * Why not Kotlinx-nodejs?
+ * -----------------------
+ *
+ * Even better than using Dukat directly would be to use the [official artifact][kotlinx_nodejs],
+ * itself generated with Dukat. We also don't use the official Node.js artifact for the reasons
+ * above, and also because it has an unstable API.
+ *
+ *
+ * Updating this file
+ * ------------------
+ *
+ * To declare new external APIs, run Dukat to generate a full set of Node stubs. The easiest way to
+ * do this is to add an NPM dependency on `@types/node` in `jsMain`, like this:
+ *
+ * ```
+ * jsMain {
+ *   ...
+ *   dependencies {
+ *     implementation(npm('@types/node', '14.14.16', true))
+ *     ...
+ *   }
+ * }
+ * ```
+ *
+ * This will create a file with a full set of APIs to copy-paste from.
+ *
+ * ```
+ * okio/build/externals/okio-parent-okio/src/fs.fs.module_node.kt
+ * ```
+ *
+ * [Dukat]: https://github.com/kotlin/dukat
+ * [kotlinx_nodejs]: https://github.com/Kotlin/kotlinx-nodejs
+ */
+@file:JsModule("fs")
+@file:JsNonModule
+package okio
+
+import org.khronos.webgl.Uint8Array
+import kotlin.js.Date
+
+internal external fun closeSync(fd: Number)
```

Comment:
`internal external fun` ?
that means we can have fun EVERYWHERE

Reply:
This has to be a tweet